### PR TITLE
[2.0]: Major refactoring

### PR DESCRIPTION
**Pull request by meatball133**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM crystallang/crystal:1.10.1-alpine
+FROM crystallang/crystal:nightly-alpine
```
Comment (ErikSchierboom) [inline Dockerfile:1]: In general we're not in favor of using nightlies, as they're not stable and can cause sudden breakages. What's the reason you need to be on the nightly track?

Comment (meatball133) [reply Dockerfile:1]: I werent planning to merge this pr, I am waiting for the 1.11.x release. That version is almost finished (bug fixes are the only thing merged at this point) so I wanted to test compatibility before the release next week.

Comment (ErikSchierboom) [reply Dockerfile:1]: Let me know when you want me to review

**Review by ErikSchierboom — APPROVED** [review]
LGTM

**Pull request merged**